x/ibc/23-commitment/types/merkle.go

```diff
@@ -128,25 +128,93 @@ func (MerkleProof) GetCommitmentType() exported.Type {
 }
 
 // VerifyMembership verifies the membership pf a merkle proof against the given root, path, and value.
-func (proof MerkleProof) VerifyMembership(root exported.Root, path exported.Path, value []byte) error {
+func (proof MerkleProof) VerifyMembership(specs []string, root exported.Root, path exported.Path, value []byte) error {
 	if proof.IsEmpty() || root == nil || root.IsEmpty() || path == nil || path.IsEmpty() || len(value) == 0 {
```

Comment:
check if specs are empty, ditto for other verification functions

Reply:
nit, we can probably move this check to a separate `func hasEmptyFields(specs, root, path) bool` function